src/client.cpp

```diff
@@ -773,11 +773,8 @@ void CClient::OnHandledSignal ( int sigNum )
     {
     case SIGINT:
     case SIGTERM:
-        // if connected, terminate connection (needed for headless mode)
-        if ( IsRunning() )
-        {
-            Stop();
-        }
+        // if connected, disconnect (needed for headless mode)
+        Disconnect();
```

Comment:
Maybe `OnAboutToQuit` should handle all this.  It feels cleaner to me to have only the `QCoreApplication::instance()->exit()` call here and deal with everything else in the handlers.

Again, if _different handlers_ are needed for UI/RPC/headless, they can be wired up appropriately on start-up.

Start up would be something like:
- Create a Client instance
- If requested, create a ClientDlg instance, passing only the pointer to the Client instance
  - Wire up Client public signals to ClientDlg private slots
  - Wire up ClientDlg private signals to Client public slots
- If requested, create a Client JSON-RPC instance, passing only the pointer to the Client instance
  - Wire up Client public signals to Client JSON-RPC private slots
  - Wire up Client JSON-RPC private signals to Client public slots
- Pass Client the start up parameters
  - effect Client start up, emitting any such signals as are needed when state changes

Thus, when exiting, it's the Client instance that's in control.  It says "OK, time to quit".  It should have itself wired up any internal Audio and (network) Channel slots to clean up the connection.  All the ClientDlg and Client JSON-RPC need to know is that exit is happening and to shut themselves down cleanly.

Reply:
> that exit is happening and to shut themselves down cleanly.

Which should then in turn also fix the connection status issue if implemented correctly.

I'll have a thought about this.

Reply:
I think it's not part of this PR though. Could you please open an issue?

Reply:
https://github.com/jamulussoftware/jamulus/issues/3402